File: src/main/java/dev/akif/cookbook/recipe/RecipeEntity.java
package dev.akif.cookbook.recipe;

import jakarta.persistence.*;
import lombok.*;

@AllArgsConstructor
@Entity(name = "recipe")
@Getter
@NoArgsConstructor
@Setter
@Table(name = "recipes")
@ToString
public class RecipeEntity {
    @Column(name = "id", nullable = false, unique = true)
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "number_of_servings", nullable = false)
    private Integer numberOfServings;
}

File: src/main/kotlin/dev/akif/cookbook/recipe/RecipeEntity.kt
package dev.akif.cookbook.recipe

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table(name = "recipes")
class RecipeEntity(
    @Id
    var id: Long? = null,
    var name: String = "",
    var numberOfServings: Int = 0
)
